SecurityPkg/Tcg/Tcg2Pei/Tcg2Pei.c

```diff
@@ -644,6 +645,64 @@ MeasureCRTMVersion (
            );
 }
 
+// MU_CHANGE [BEGIN]
+/**
+Measure Insecure Device State into PCR 07 and PCR 04
+
+@retval EFI_SUCCESS           Operation completed successfully.
+@retval EFI_OUT_OF_RESOURCES  Not enough memory to log the new event.
+@retval EFI_DEVICE_ERROR      The command was unsuccessful.
+
+**/
+EFI_STATUS
+MeasureInsecureDeviceState (
+  VOID
+  )
+{
+  CHAR8              *String;
+  TCG_PCR_EVENT_HDR  TcgEventHdr;
+  EFI_STATUS         Status;
+
+  //
+  // Use FirmwareVersion string to represent CRTM version.
+  // OEMs should get real CRTM version string and measure it.
+  //
+
+  String                = MSFT_INSECURE_DEVICE_STATE_STRING;
+  TcgEventHdr.PCRIndex  = 7;
+  TcgEventHdr.EventType = EV_EFI_ACTION;
+  TcgEventHdr.EventSize = (UINT32)sizeof (MSFT_INSECURE_DEVICE_STATE_STRING);
+
+  Status = HashLogExtendEvent (
+           &mEdkiiTcgPpi,
+           0,
+           (UINT8 *)String,
+           TcgEventHdr.EventSize,
+           &TcgEventHdr,
+           (UINT8 *)String
+           );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "Failed to measure Insecure Device State to PCR 07\n"));

```

Comment:
Might as well throw an assert in here. We could consider panicking, too. @cfernald @makubacki thoughts?

Reply:
Should we return from the function or still try to measure into PCR 04?

Reply:
Talked offline, determined we will panic and not cap the PCR as it's not a PCR-capping error scenario defined in the specification.